ath79: add support for 8devices Lima board

andreybondar59 opened the pull request:
Specification:

 - 650/600/216 MHz (CPU/DDR/AHB)
 - 64 MB of RAM (DDR2)
 - 32 MB of FLASH
 - 2T2R 2.4 GHz
 - 2x 10/100 Mbps Ethernet
 - 1x USB 2.0 Host socket
 - 1x miniPCIe slot
 - UART for serial console
 - 14x GPIO

Flash instructions:

Upgrading from ar71xx target:
 - Upload image into the board:
    scp openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin \
        root@192.168.1.1/tmp/
 - Run sysupgrade
    sysupgrade -F /tmp/openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin

Upgrading from u-boot:
 - Set up tftp server with openwrt-ath79-generic-8dev_lima-initramfs-kernel.bin
 - Go to u-boot (reboot and press ESC when prompted)
 - Set TFTP server IP
    setenv serverip 192.168.1.254
 - Set device ip from the same subnet
    setenv ipaddr 192.168.1.1
 - Copy new firmware to board
    tftpboot 0x82000000 initramfs.bin
 - Boot OpenWRT
    bootm 0x82000000
 - Upload image openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin into the board
 - Run sysupgrade.

Signed-off-by: Andrey Bondar <a.bondar@8devices.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+		bootargs="console=ttyS0,115200n81";
+        };
+
+	keys {
```
Please check your indent. DTS should be tabs-only.

andreybondar59 replied:
Fixed.

adrianschmutzler replied:
Not fixed in a lot of places in this file.

andreybondar59 replied:
Fixed.

adrianschmutzler replied:
No.

adrianschmutzler replied:
Ah, now.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		button@0 {
```
In OpenWrt, it's common to use the label as node name, i.e. just "reset {" here.

andreybondar59 replied:
Fixed.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+         status = "okay";
+
+         /* Winbond W25Q256 SPI flash */
+         spiflash: w25q256@0 {
```
Generic names are preferred here, i.e. flash@0

andreybondar59 replied:
Fixed.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+         spiflash: w25q256@0 {
+                #address-cells = <1>;
+                #size-cells = <1>;
+                compatible = "jedec,spi-nor", "winbond,w25q256";
```
Is the second string required?

andreybondar59 replied:
Not required, fixed.

adrianschmutzler replied:
Not fixed.

andreybondar59 replied:
Fixed.

adrianschmutzler replied:
No, again.

adrianschmutzler commented on target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:
```diff
@@ -9,6 +9,9 @@ board=$(board_name)
 	case $board in
+	8dev,lima)
+		caldata_extract "art" 0x1000 0x800
+		;;
```
This can be merged with ubnt,unifi (i.e. the latter should be moved up here, too).

andreybondar59 replied:
Merged.

adrianschmutzler commented on target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:
```diff
@@ -43,6 +46,10 @@ case "$FIRMWARE" in
+	enterasys,ws-ap3705i)
+		caldata_extract "calibrate" 0x1000 0x440
+		ath9k_patch_mac $(mtd_get_mac_ascii u-boot-env0 RADIOADDR1)
+		;;
```
Does that belong here?

andreybondar59 replied:
Not quite clear to me. What do you mean?

adrianschmutzler replied:
You originally copied unrelated stuff for some other device here. That seems to have been resolved in the meantime as it appears.

adrianschmutzler commented on target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:
```diff
@@ -72,6 +79,10 @@ case "$FIRMWARE" in
+	enterasys,ws-ap3705i)
+		caldata_extract "calibrate" 0x5000 0x440
+		ath9k_patch_mac $(mtd_get_mac_ascii u-boot-env0 RADIOADDR0)
+		;;
```
Does that belong here?

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -304,6 +313,16 @@ define Device/comfast_cf-e120a-v3
+  IMAGE_SIZE := 7936k
+endef
+TARGET_DEVICES += comfast_cf-e130n-v2
+
```
Does that belong here?

andreybondar59 replied:
Fixed.

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -630,6 +649,14 @@ define Device/engenius_ews511ap
+  IMAGE_SIZE := 30528k
+endef
+TARGET_DEVICES += enterasys_ws-ap3705i
+
```
Does that belong here?

adrianschmutzler commented:
Does this use the default case in 02_network by intention.

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -116,6 +115,16 @@ define Device/8dev_carambola2
+  DEVICE_VENDOR := 8devices
+  DEVICE_MODEL := Lima
+  DEVICE_PACKAGES := kmod-usb2
+  IMAGE_SIZE := 15616k
```
That doesn't match your partitioning at all ...

andreybondar59 replied:
Fixed to value 0xf40000 in dts file.

andreybondar59 commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+	keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
```
Fixed.

andreybondar59 commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+};
+
+&eth1 {
+	status = "okay";
```
Ok, fixed.

andreybondar59 commented on target/linux/ath79/image/generic.mk:
```diff
@@ -116,6 +115,16 @@ define Device/8dev_carambola2
+  DEVICE_MODEL := Lima
+  DEVICE_PACKAGES := kmod-usb2
+  IMAGE_SIZE := 15616k
+  SUPPORTED_DEVICES += lima
```
OEM firmware is based on OpenWrt.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,120 @@
+        model = "8devices Lima";
+
+        chosen {
+		bootargs="console=ttyS0,115200n81";
```
In case you mean n8 instead of n81, the chosen node can be dropped entirely due to a recent patch on master.

andreybondar59 replied:
Fixed.

ynezz commented:
Please address the comments, resolve the merge conflict, add missing commit description.

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -118,7 +118,6 @@ define Device/seama
   SEAMA_SIGNATURE :=
 endef
 
-
```
Don't touch that line.

andreybondar59 commented on target/linux/ath79/dts/qca9531_8dev_lima.dts:
```diff
@@ -0,0 +1,123 @@
+				read-only;
+			};
+
+			ubootenv: partition@40000 {
```
Dropped.

andreybondar59 commented:
Sorry, didn't commit at a time.

On Wed, Jul 8, 2020 at 11:37 AM Adrian Schmutzler <notifications@github.com>
wrote:

> *@adrianschmutzler* commented on this pull request.
> ------------------------------
>
> In target/linux/ath79/dts/qca9531_8dev_lima.dts
> <https://github.com/openwrt/openwrt/pull/3078#discussion_r451377866>:
>
> > +	status = "okay";
> +};
> +
> +&wdt {
> +         status = "okay";
> +};
> +
> +&spi {
> +         num-cs = <1>;
> +         status = "okay";
> +
> +         /* Winbond W25Q256 SPI flash */
> +         spiflash: w25q256@0 {
> +                #address-cells = <1>;
> +                #size-cells = <1>;
> +                compatible = "jedec,spi-nor", "winbond,w25q256";
>
> No, again.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3078#discussion_r451377866>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOG5KUJHA4D2WARPM4BGI3R2QV5BANCNFSM4NSVT4OQ>
> .
>

adrianschmutzler commented:
Still need to squash your two commits into one.

andreybondar59 commented:
Thanks for your patience. Squashed in one.

adrianschmutzler commented:
And now you need a commit description, i.e. probably the content of your first post.

andreybondar59 committed:
ath79: add support for 8devices Lima board

Specification:

  • 650/600/216 MHz (CPU/DDR/AHB)
  • 64 MB of RAM (DDR2)
  • 32 MB of FLASH
  • 2T2R 2.4 GHz
  • 2x 10/100 Mbps Ethernet
  • 1x USB 2.0 Host socket
  • 1x miniPCIe slot
  • UART for serial console
  • 14x GPIO

Flash instructions:

Upgrading from ar71xx target:

  • Upload image into the board:
    scp openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin
    root@192.168.1.1/tmp/
  • Run sysupgrade
    sysupgrade -F /tmp/openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin

Upgrading from u-boot:

  • Set up tftp server with
    openwrt-ath79-generic-8dev_lima-initramfs-kernel.bin
  • Go to u-boot (reboot and press ESC when prompted)
  • Set TFTP server IP
    setenv serverip 192.168.1.254
  • Set device ip from the same subnet
    setenv ipaddr 192.168.1.1
  • Copy new firmware to board
    tftpboot 0x82000000 initramfs.bin
  • Boot OpenWRT
    bootm 0x82000000
  • Upload image openwrt-ath79-generic-8dev_lima-squashfs-sysupgrade.bin into
    the board
  • Run sysupgrade.

Signed-off-by: Andrey Bondar <a.bondar@8devices.com>

andreybondar59 commented:
Done.

On Wed, Jul 8, 2020 at 1:12 PM Adrian Schmutzler <notifications@github.com>
wrote:

> And now you need a commit description, i.e. probably the content of your
> first post.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3078#issuecomment-655426367>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOG5KVZTLR2BRKXAIQINWTR2RA67ANCNFSM4NSVT4OQ>
> .
>

adrianschmutzler commented:
@ynezz Might be worth another look from you now.

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git
